sktime/param_est/gcf.py

```diff
@@ -0,0 +1,420 @@
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Granger causality parameter estimator."""
+
+__author__ = ["Spinachboul"]
+__all__ = ["GrangerCausalityFitter"]
+
+import pandas as pd
+
+from sktime.exceptions import NotFittedError
+from sktime.param_est.base import BaseParamFitter
+from sktime.utils.dependencies._dependencies import _check_soft_dependencies
+
+# Import statsmodels functions conditionally
+_check_soft_dependencies("statsmodels", severity="warning")
```

Comment:
can you move the imports to inside the class?

That should avoid doing the check manually.